app/models/user.rb

```diff
@@ -1,12 +1,17 @@
 class User < ApplicationRecord
-  has_many :subjects#, dependent: destroy
-  has_many :user_has_subjects#, dependent: destroy
+  # has_and_belongs_to_many :subjects#, through: :user_has_subject #, dependent: destroy
+  # has_many :subjects, through: :user_has_subject
+  has_many :active_subjects, class_name: "UserHasSubject",
+                             foreign_key: :user_id, 
+                             dependent: :destroy
   has_many :exams#, dependent: destroy
   has_many :trainee_answer_sheets#, dependent: destroy
   has_many :questions
   # has_many :users
   # belongs_to :user, foreign_key: :create_by
   enum role: [:trainee, :admin, :supervisor] 
 
+  scope :trainees,-> {where role: User.roles[:trainee]}
```

Comment:
cần scope ko chú

`User.tranee` thì có lấy dc toàn bộ user có role là trainee ko
tìm hiểu thêm các method dc hỗ trợ bởi enum nào